HeartbeatLogging/Tests/Unit/HeartbeatStorageTests.swift

```diff
@@ -285,8 +285,10 @@ private class StorageFake: Storage {
   func read() throws -> Data {
     if let onRead = onRead {
       return try onRead()
+    } else if let data = data {
+      return data
```

Comment:
Separate from this PR: 

- how can `data` ever not be nil?
- Why doesn't `onRead` return data?
- Why is data at class scope?

Reply:
> how can data ever not be nil?
Data can be given a non-nil value by using the StorageFake's `write()` API.

Consider:
```swift
let storageFake = StorageFake() // `data` == `nil`
let myData: Data = // ...
storageFake.write(myData) // `data` == myData
```

Data may not be nil when non-nil data is  nilwrite()

> Why doesn't onRead return data?
`onRead` _does_ return data. It is an optional, 
throwing closure that returns data: `(() throws -> Data)?`

> Why is data at class scope?
Hmm it is not, `data` is an instance var.

Reply:
It's not a class var, but it's an instance var at class scope versus locally scoped to a `func`.

Why does `write` need to store data in `self.data` in addition to passing it to `onWrite`?
Why does `read` ignore the return value from `onRead` and instead return what was previously written by `write`?

Reply:
Done in [515633d](https://github.com/firebase/firebase-ios-sdk/pull/9361/commits/515633dbab7dadaa76962f64d66d803fee1cc61f)